internal/pkg/filtersBlockService/FilterMAIR.go

```diff
@@ -0,0 +1,166 @@
+// FilterMAIR implements a trimmed moving average. Outliers are eliminated using interquartile range
+// see: https://en.wikipedia.org/wiki/Interquartile_range
+package filters
+
+import (
+	"github.com/diadata-org/diadata/pkg/dia"
+	"github.com/diadata-org/diadata/pkg/model"
+	log "github.com/sirupsen/logrus"
+	"math"
+	"sort"
+	"strconv"
+	"time"
+)
+
+// FilterMAIR contains the configuration parameters of the filter
+type FilterMAIR struct {
+	symbol         string
+	exchange       string
+	currentTime    time.Time
+	previousPrices []float64
+	lastTrade      *dia.Trade
+	memory         int
+	value          float64
+}
+
+//NewFilterMAIR creates a FilterMAIR
+func NewFilterMAIR(symbol string, exchange string, currentTime time.Time, memory int) *FilterMAIR {
+	s := &FilterMAIR{
+		symbol:         symbol,
+		exchange:       exchange,
+		previousPrices: []float64{},
+		currentTime:    currentTime,
+		memory:         memory,
+	}
+	return s
+}
+func computeMean(samples []float64) (mean float64) {
+	var total float64
+	length := float64(len(samples))
+	if length == 0 {
+		return
+	}
+	for _, s := range samples {
+		total += s
+	}
+	mean = total / length
+	return
+}
+func computeMedian(samples []float64) (median float64) {
+	var length = len(samples)
+	if length > 0 {
+		sort.Float64s(samples)
+		if length%2 == 0 {
+			median = (samples[length/2-1] + samples[length/2]) / 2
+		} else {
+			median = samples[(length+1)/2-1]
+		}
+	}
+	return
+}
+func computeQuartiles(samples []float64) (Q1 float64, Q3 float64) {
+	sort.Float64s(samples)
+	var length = len(samples)
+	if length > 0 {
+		if length%2 == 0 {
+			Q1 = computeMedian(samples[0 : length/2])
+			Q3 = computeMedian(samples[length/2 : length])
+		} else {
+			Q1 = computeMedian(samples[0:int(math.Floor(float64(length/2)))])
```

Comment:
do we really need the math.floor here ? since we know it's odd, can we do like in computeMedian ?